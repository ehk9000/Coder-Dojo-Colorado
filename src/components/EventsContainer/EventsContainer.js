import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import requests from '../../utils/requests/requests';
import EventCard from '../EventCard/EventCard';

export function EventsContainer(props) {
  const { match, isLoading, error } = props;
  const [upcomingEvents, setUpcomingEvents] = useState([]);
  const [recentEvents, setRecentEvents] = useState([]);
  const [venue, setVenue] = useState({});
  
  const getAllData = async (venueId) => {
    const upcoming = await requests.getUpcomingEvents({ venueId });
    const recent = await requests.getRecentEvents({ venueId });
    const currentVenue = await requests.getVenueDetails({ venueId });
    setUpcomingEvents(upcoming.futureEvents);
    setRecentEvents(recent.pastEvents);
    setVenue(currentVenue.allVenues[0]);
  };

  useEffect(() => {
    getAllData(parseInt(match.params.id, 10));
  }, [match]);

  const generateEvents = events => events.map(event => <EventCard key={event.id} event={event} />);

  const venueDetails = () => {
    const { 
      city, street1, street2, zip, state
    } = venue.addresses[0];
    const venueAdress = `${street1}, ${street2 || ''} ${city}, ${state} ${zip}`;
    return (
      <section className="venue-details">
        <p>{venue.notes}</p>
        <h4>Address:</h4>
        <p>
          {venueAdress}
        </p>
      </section>
    );
  };

  return (
    <section className="EventsContainer">
      <div className="venue-header">
        <h2>{venue.name}</h2>
        { isLoading && <p>Loading events...</p> }
      </div>
      <div className="details-container">
        {error && <p>{error}</p>}
        {venue.name && venueDetails()}
        <section className="events-container">
          <h3>Upcoming Events:</h3>
          {upcomingEvents.length > 0 && generateEvents(upcomingEvents)}
          <h3>Recent Events:</h3>
          {recentEvents.length > 0 && generateEvents(recentEvents)}
        </section>
      </div>
    </section>
  );
}

export const mapStateToProps = state => ({
  isLoading: state.isFetching,
  error: state.error
});

export default connect(mapStateToProps)(EventsContainer);

EventsContainer.propTypes = {
  match: PropTypes.object,
  isLoading: PropTypes.bool,
  error: PropTypes.string
};

EventsContainer.defaultProps = {
  match: {},
  isLoading: false,
  error: ''
};
