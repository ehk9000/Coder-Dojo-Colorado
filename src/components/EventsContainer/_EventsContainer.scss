.EventsContainer {
  width: 80%;
  margin: auto;
  height: 100vh;
  
  .venue-header {
    width: 100%;
    padding: 30px 0;
    display: flex;
    align-items: center;
    background-color: #724695;
    color: #fff;

    h2 {
      font-weight: 400;
      margin: 0 40px;
    }
  }

  .details-container {
    display: flex;
    height: 100%;
    background-color: #E5E5E5;

    .venue-details {
      flex: 2;
      padding: 20px;
    }
    
    .events-container {
      flex: 5;
      background-color: #fff;
      padding: 15px;
    }
  }
}