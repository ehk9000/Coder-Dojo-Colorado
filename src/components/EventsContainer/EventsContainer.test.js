import React from 'react';
import { mount } from 'enzyme';
import { EventsContainer } from './EventsContainer';

describe('EventsContainer', () => {
  let wrapper;
  const mockError = '';
  const mockMatch = {
    params: {
      id: 1
    }
  };

  beforeEach(() => {
    wrapper = mount(<EventsContainer match={mockMatch} isLoading={false} error={mockError} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
