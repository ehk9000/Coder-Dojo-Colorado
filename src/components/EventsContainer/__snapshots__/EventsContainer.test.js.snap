// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsContainer should match snapshot 1`] = `
<EventsContainer
  error=""
  isLoading={false}
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
>
  <section
    className="EventsContainer"
  >
    <div
      className="venue-header"
    >
      <h2 />
    </div>
    <div
      className="details-container"
    >
      <section
        className="events-container"
      >
        <h3>
          Upcoming Events:
        </h3>
        <h3>
          Recent Events:
        </h3>
      </section>
    </div>
  </section>
</EventsContainer>
`;
